### ISSUE TITLE
Rule Proposal: `padding-line-between-sibling-components`

### ISSUE THREAD

**Issue by Evgenios95** [vuejs/eslint-plugin-vue#1834]
**Please describe what the rule should do:**

This rule should enforce a padding line between **sibling** components in `.vue` files and throw a warning or error in case the convention is not followed. 

**What category should the rule belong to?**

[X] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<!-- RECOMMENDED -->
</template>
   <div class="whatever">
      <slot />

      <hello-world-component />

      <icon-component />
    </div>
</template>

<!-- NOT RECOMMENDED -->
</template>
   <div class="whatever">
      <slot />
      <hello-world-component />
      <icon-component />
    </div>
</template>
```

```vue
<!-- RECOMMENDED -->
</template>
   <div class="div-element" >
      <h1 class="h1-element" />
      <p class="p-element" />
      <div class="components-wrapper">
         <slot />

         <hello-world-component />
      </div>
    </div>
</template>

<!-- NOT RECOMMENDED -->
</template>
   <div class="div-element" >
      <h1 class="h1-element" />
      <p class="p-element" />
      <div class="components-wrapper">
         <slot />
         <hello-world-component />
      </div>
    </div>
</template>
```

**Additional Context**:

Similar to vue/padding-line-between-blocks https://eslint.vuejs.org/rules/padding-line-between-blocks.html

**Comment by ota-meshi** [comment]
Duplicate of #1832

**Comment by Evgenios95** [comment]
@ota-meshi He suggests tags in general, I asked specifically about components.

**Comment by ota-meshi** [comment]
Thank you for clarifying.
I think it's better to create one rule than to create the rule separately for the components and the HTML elements.
Perhaps there is no benefit to having separate rules.
